### ISSUE TITLE
Add tagging to generic Viewer interface

### ISSUE THREAD

**Issue by ajratner** [HazyResearch/snorkel#231]
After #230 , next step then will be to add a way to tag `Annotation` objects (Enter / button) and save these tags to the annotation object; will implement just the above `Viewer`

**Comment by henryre** [comment]
Would be cool to have something beyond T/F so that people can come back and look at problem cases

**Comment by ajratner** [comment]
Agreed.  My goal is just to have a very simple, generic interface.  So rough proposal:

* In addition to supplying a set of `Context` objects and corresponding sets of `Annotation` objects (I know this will need to be worked out further...), they can also supply a `tag_attribute` which is the name of an attribute of the each `Annotation` object that can be modified- this is where tags will be saved.

* Additionally, the user can provide two things:
  - A list of possible tags (`default=[True, False]`)
  - A tag checker (`default=lambda tag : isinstance(tag, bool)`)

...and then this will generate some autocomplete list that pops up when Enter is clicked (will just take one of the existing js libs out there that does this)

**Comment by ajratner** [comment]
Again, also remember that the explicit goal here is to make a simple _default_ interface that is (a) simple and (b) easily modifiable, and that can exist in addition to 'power tools' with more functinality

**Comment by henryre** [comment]
Nice v similar to what I was thinking! Looking ahead a bit, for the actual tagging app, we probably do want to make T/F "special" tags and provide one-click buttons for them.

**Comment by ajratner** [comment]
T/F implemented in merged PR; closing this, we can open new issue for non-boolean tags if needed
